Refactor manual install docs

lulf opened the pull request:
* Use the term namespace which is a bit more precise in the context of
deploying singletenant
* Clarify prerequisites for deploying to a single namespace
* Fix references to manual steps so that Kubernetes and OpenShift steps
are aligned
* Delete unused old doc files

ncbaratta commented on documentation/service_admin/proc-deploying-address-space-controller.adoc:
```diff
@@ -9,7 +9,7 @@ include::common/attributes.adoc[]
 The address space controller is responsible for creating the infrastructure used by address spaces.
 
-NOTE: To install {ProductName} on OpenShift, you must have `cluster-admin` access to set up the required roles for creating namespaces and managing resources in those namespaces; otherwise, you are restricted to a single address space. For more information about how to deploy without `cluster-admin` access, which restricts {ProductName} to a single address space, see xref:deploying-single-address ...
+NOTE: To install {ProductName} on OpenShift, you must have `cluster-admin` access to set up the required roles for creating namespaces and managing resources in those namespaces; otherwise, you are restricted to a single address space. For more information about how to deploy without `cluster-admin` access, which restricts {ProductName} to a single address space, see xref:deploying-single-address ...
```
I think this was right before.  You want to put the link to the procedure in question

lulf replied:
The problem her is that `proc-deploying-address-space-controller` is part of `assembly-installing-manual-steps.adoc` which is part of both `assembly-installing-openshift` and `assembly-installing-kubernetes` which are 2 different contexts. This means that it needs to be linked to in 2 different contexts as well.

In the final doc, the kubernetes doc will end up linking to deploying-single-address-space-kube, and the openshift doc will end up linking to deploying-single-address-space-maas.

You can see the end result here: 

http://people.redhat.com/~ulilleen/maasdoc/master.html#deploying-address-space-controller-maas which links to http://people.redhat.com/~ulilleen/maasdoc/master.html#deploying-single-address-space-maas

http://people.redhat.com/~ulilleen/maasdoc/master.html#deploying-address-space-controller-kube which links to http://people.redhat.com/~ulilleen/maasdoc/master.html#deploying-single-address-space-kube

Am I using 'context' for the wrong thing here, and should instead create a different parameter?

On a separate note, I was wondering that if one procedure needs to be referenced in different contexts within the 'enmasse' documentation _and_ across products, the context needs to be 'appended', not overwritten. I.e. we should do `:context: {context}-maas` instead of `:context: maas` when assigning it.

lulf replied:
@ncbaratta Reading your other comment, I get the impression that as you say context should only be assigned in the master adoc, and not in the individual files. This means then that in order to have different contexts within the doc as we need for the installation docs, we should choose a different parameter.

ncbaratta commented on documentation/service_admin/proc-deploying-single-namespace.adoc:
```diff
@@ -5,7 +5,14 @@
+
+.Prerequisite
+
+* At least one authentication service must be installed, see xref:deploying-auth-services-{context}[].
```
Also I don't think this should be {context} it should be whatever context is on the page you're linking to.

lulf commented:
@ncbaratta @jenmalloy I've updated the PR with the changes I believe are not interferring with the use of `context` anymore. Please have another look.

The 'problem' here is that we have things like the 'manual install' documentation which is included twice, once for Kubernetes and once for OpenShift. The assemblies and procedures that make up the manual install docs need to have the id's and links parameterized by something. This is to avoid having to write them twice. I've changed the docs to use a 'manual-install-context' variable to achieve the same thing without messing with the 'context' parameter.

I've went through the Strimzi docs on master and I didn't actually find  a similar pattern of reuse between Kubernetes and OpenShift docs. Please point me to it in case I've overlooked it so that I can learn from the example.

jenmalloy commented on documentation/service_admin/assembly-installing-openshift.adoc:
```diff
@@ -2,6 +2,9 @@
 :parent-context: {context}
+:cmdcli: oc
+:KubePlatform: OpenShift
+:manual-install-context: okd
```
Should be: 
:context: messaging

jenmalloy commented on documentation/service_admin/proc-deploying-address-space-controller.adoc:
```diff
@@ -4,12 +4,12 @@
 include::common/attributes.adoc[]
 
-[id='deploying-address-space-controller-{context}']
+[id='deploying-address-space-controller-{manual-install-context}-{context}']
```
Delete new content

jenmalloy commented on documentation/service_admin/proc-deploying-api-server.adoc:
```diff
@@ -4,13 +4,16 @@
 include::common/attributes.adoc[]
 
-[id='deploying-api-server-{context}']
+[id='deploying-api-server-{manual-install-context}-{context}']
```
Delete new content

jenmalloy commented on documentation/service_admin/proc-deploying-auth-services.adoc:
```diff
@@ -4,7 +4,7 @@
 include::common/attributes.adoc[]
 
-[id='deploying-auth-services-{context}']
+[id='deploying-auth-services-{manual-install-context}-{context}']
```
Delete new content

jenmalloy commented on documentation/service_admin/proc-deploying-address-space-controller.adoc:
```diff
@@ -4,12 +4,12 @@
 The address space controller is responsible for creating the infrastructure used by address spaces.
 
-NOTE: To install {ProductName} on OpenShift, you must have `cluster-admin` access to set up the required roles for creating namespaces and managing resources in those namespaces; otherwise, you are restricted to a single address space. For more information about how to deploy without `cluster-admin` access, which restricts {ProductName} to a single address space, see xref:deploying-single-address ...
+NOTE: To install {ProductName} on {KubePlatform} , you must have `cluster-admin` access to set up the required roles for creating namespaces and managing resources in those namespaces; otherwise, you are restricted to a single address space. For more information about how to deploy without `cluster-admin` access, which restricts {ProductName} to a single address space, see xref:deploying-single-n ...
```
{PlatformName}
Delete "{manual-install-context}"

jenmalloy commented on documentation/service_admin/assembly-installing-manual-steps.adoc:
```diff
@@ -1,15 +1,14 @@
-:cmdcli: oc
 
-[id='installing-using-manual-steps-{context}']
+[id='installing-using-manual-steps-{manual-install-context}-{context}']
```
Delete new content

jenmalloy commented on documentation/service_admin/assembly-installing-manual-steps.adoc:
```diff
@@ -1,15 +1,14 @@
 
 The manual deployment procedure can be performed on any platform
-supporting the OpenShift client.
+supporting the {KubePlatform} client.
```
{PlatformName}

jenmalloy commented on documentation/service_admin/proc-creating-project.adoc:
```diff
@@ -5,7 +5,7 @@
 include::common/attributes.adoc[]
 
-[id='creating-project-{context}']
+[id='creating-project-{manual-install-context}-{context}']
```
Delete new content

jenmalloy commented on documentation/service_admin/proc-deploying-api-server.adoc:
```diff
@@ -4,13 +4,16 @@
 a Kubernetes API server if it is registered as an APIService.
 
-NOTE: To install {ProductName} on OpenShift, you must have `cluster-admin` access to set up the required roles for delegating authentication to the Kubernetes master; otherwise, you are restricted to a single address space. For more information about how to deploy without `cluster-admin` access, which restricts {ProductName} to a single address space, see xref:deploying-single-address-space-messa ...
+NOTE: To install {ProductName} on {KubePlatform}, you must have `cluster-admin` access to set up the
```
{PlatformName}

jenmalloy commented on documentation/service_admin/proc-deploying-api-server.adoc:
```diff
@@ -4,13 +4,16 @@
+NOTE: To install {ProductName} on {KubePlatform}, you must have `cluster-admin` access to set up the
+required roles for delegating authentication to the Kubernetes master; otherwise, you are restricted
+to a single address space. For more information about how to deploy without `cluster-admin` access,
+which restricts {ProductName} to a single address space, see xref:deploying-single-namespace-{manual-install-context}-{context}[].
```
Delete "{manual-install-context}"

jenmalloy commented on documentation/service_admin/proc-deploying-service-broker.adoc:
```diff
@@ -2,12 +2,12 @@
 // assembly-installing-manual-steps.adoc
 
-[id='deploying-service-broker-{context}']
+[id='deploying-service-broker-{manual-install-context}-{context}']
```
Delete new content

jenmalloy commented on documentation/service_admin/proc-deploying-service-broker.adoc:
```diff
@@ -2,12 +2,12 @@
 The service broker provides an implementation of the Open Service Broker API that integrates with the Kubernetes Service Catalog. The service broker requires the standard authentication service to be deployed.
 
-NOTE: To install {ProductName} on OpenShift, you must have `cluster-admin` access to set up the required roles for delegating authentication to the Kubernetes master; otherwise, you are restricted to a single address space. For more information about how to deploy without `cluster-admin` access, which restricts {ProductName} to a single address space, see xref:deploying-single-address-space-messa ...
+NOTE: To install {ProductName} on {KubePlatform}, you must have `cluster-admin` access to set up the required roles for delegating authentication to the Kubernetes master; otherwise, you are restricted to a single address space. For more information about how to deploy without `cluster-admin` access, which restricts {ProductName} to a single address space, see xref:deploying-single-namespace-{man ...
```
{PlatformName}
Delete "{manual-install-context}"

jenmalloy commented on documentation/service_admin/proc-deploying-single-namespace.adoc:
```diff
@@ -4,8 +4,15 @@
 
-[id='deploying-single-address-space-{context}']
-= Deploying {ProductName} to a single address space
+[id='deploying-single-namespace-{manual-install-context}-{context}']
```
Delete "{manual-install-context}"

jenmalloy commented on documentation/service_admin/proc-deploying-single-namespace.adoc:
```diff
@@ -4,8 +4,15 @@
+
+.Prerequisite
+
+* At least one authentication service must be installed. For more information see * xref:deploying-auth-services-{manual-install-context}-{context}[].
```
Delete "{manual-install-context}"

lulf commented:
@jenmalloy Pushed an update with fixes.
